### PR TITLE
[AUTO] Load cached model to target device W/O CPU accelerating

### PR DESCRIPTION
**Pull request by yangwang201911**
### Details:
 - enable AUTO to retrieve the hash id for cached model
 - load cached model to GPU  only with AUTO if cached blob exists and not set `ov::intel_auto::enable_startup_fallback` explicitly to AUTO

### Tickets:
 - CVS-140059

### CODE REVIEW

File: src/plugins/auto/src/auto_schedule.cpp [hunk 1]
```diff
@@ -139,10 +142,50 @@ void AutoSchedule::init() {
...
+                    } else {
+                        // user set the compiling threads num
+                        // use the user's val anyway
+                        LOG_DEBUG_TAG("user defined compiling threads: %d", max_threads);
+                    }
+                    std::string cache_dir = device_config.count(ov::cache_dir.name())
+                                                ? device_config[ov::cache_dir.name()].as<std::string>()
+                                                : m_context->m_ov_core->get_property("", ov::cache_dir);
+
+                    if (!m_context->m_is_set_startup_fallback && !cache_dir.empty()) {
```
Comment (wangleis) [inline src/plugins/auto/src/auto_schedule.cpp:170]: This logic applies not only to GPUs but to all HW accelerator, for example, AUTO:NPU,CPU

Comment (yangwang201911) [reply src/plugins/auto/src/auto_schedule.cpp:172]: Updated.

File: src/plugins/auto/src/auto_schedule.cpp [hunk 2]
```diff
@@ -139,10 +142,52 @@ void AutoSchedule::init() {
...
+                            // user set the compiling threads num
+                            // use the user's val anyway
+                            LOG_DEBUG_TAG("user defined compiling threads: %d", max_threads);
+                        }
+                    }
+                    std::string cache_dir = device_config.count(ov::cache_dir.name())
+                                                ? device_config[ov::cache_dir.name()].as<std::string>()
+                                                : m_context->m_ov_core->get_property("", ov::cache_dir);
+
+                    if (!m_context->m_is_set_startup_fallback && !cache_dir.empty()) {
```
Comment (wangleis) [inline src/plugins/auto/src/auto_schedule.cpp:172]:
```suggestion
                    if (m_context->m_is_set_startup_fallback && !cache_dir.empty()) {
```

File: src/plugins/auto/src/auto_schedule.cpp [hunk 3]
```diff
@@ -139,10 +143,55 @@ void AutoSchedule::init() {
...
+                if (!is_actual_cpu) {
+                    // user does not set the compiling threads
+                    // limit the threads num for compiling
+                    auto device = m_compile_context[ACTUALDEVICE].m_device_info.device_name;
+                    auto& device_config = m_compile_context[ACTUALDEVICE].m_device_info.config;
+                    if (is_actual_gpu) {
+                        int max_threads = 0;
+                        try {
+                            max_threads = m_context->m_ov_core->get_property(device, ov::compilation_num_threads);
+                        } catch (const ov::Exception&) {
```
Comment (songbell) [inline src/plugins/auto/src/auto_schedule.cpp:155]: why set here？ we already have compilation threads set here?
https://github.com/openvinotoolkit/openvino/blob/7f784509f3b791a2bb41c787c073fb9caf7f65b8/src/plugins/auto/src/auto_schedule.cpp#L320

Comment (yangwang201911) [reply src/plugins/auto/src/auto_schedule.cpp:155]: AUTO plugin need to use the same config of actual device as CORE level to ensure the consistent  retrieval of model hash id. So need to move the config setting here.

Comment (songbell) [reply src/plugins/auto/src/auto_schedule.cpp:155]: and , compilation num threads is not a property that affects caching?
image

Comment (yangwang201911) [reply src/plugins/auto/src/auto_schedule.cpp:155]: You are right. this property did not affect caching hash ID. Will correct this logic at the next PR.

**Review by songbell — APPROVED** [review]

**Review by wangleis — APPROVED** [review]

**Pull request merged**